manifests/kustomize/base/metadata/execution-cache-deployer-clusterrole.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```

Comment:
qq - does the deployer has to be ran with a clusterrole?

Reply:
Yes. For generating certification request it needs cluster-scope access.

Reply:
will it calls external host to get certification?

Reply:
I'm a bit concerned about the minimum permissions needed to enable this feature. 

In the past, we deliberately avoid clusterrole whenever possible and so far we don't have any if I remember correctly. things are running using roles instead. There are users explicitly express their limitations and pains of deploy cluster level resources (requires escalated process setting up things)

If the communication is always inside the cluster within single namespace , I'm wondering if a certificate is necessary. I could be wrong on this since I didn't do much of research.

Reply:
I'm also a little concerned after reading this: https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#use-caution-when-authoring-and-installing-mutating-webhooks.

I worry this may limit how many users can actually use the mutation webhook. I'm not too worried though, because mutating webhook looks like an implementation detail. If we did find an alternative, the caching system would still work. We just need a different way to adjust runtime behavior.

Not went through enough details, would https://github.com/argoproj/argo/blob/master/examples/pod-spec-patch.yaml#L13 combined with init containers be able to do what you want with caching?

e.g. a pod can update its own label by using the following in one of its init container
```
cat > patch.json <<EOF
[                                                                                  {                                                                                 "op": "add", "path": "/metadata/labels/hello", "value": "world"
 }
]
EOF
curl -v --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" --request PATCH --data "$(cat patch.json)" -H "Content-Type:application/json-patch+json" https://kubernetes.default.svc/api/v1/namespaces/$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)/pods/$HOSTNAME
```